middle_end/flambda2/simplify_shared/slot_offsets.ml

```diff
@@ -203,54 +203,56 @@ let layout_aux j slot (startenv, acc_slots) =
        function, so they should not appear. *)
     assert false
 
-let layout env closures env_vars =
+let layout env function_slots value_slots =
   let map =
     Numeric_types.Int.Map.empty
-    |> order_env_vars env env_vars
-    |> order_closures env closures
+    |> order_value_slots env value_slots
+    |> order_function_slots env function_slots
   in
   let startenv_opt, acc_slots =
     Numeric_types.Int.Map.fold layout_aux map (None, [])
   in
   let startenv, empty_env =
-    (* If there are no env vars, the start of env is considered to be the
+    (* If there are no value slots, the start of env is considered to be the
        (non-existing) slot after the last slot used, and if the set is empty,
        the value does not matter. *)
     match startenv_opt, acc_slots with
     | Some i, _ -> i, false
     | None, [] -> 0, true
-    | None, (j, Closure closure) :: _ -> begin
-      match EO.closure_offset env closure with
-      | Some (Id_slot { size; _ }) -> j + size, true
-      | Some Dead_id | None ->
-        (* the closure was found earlier during the call to order_closures *)
+    | None, (j, Function_slot function_slot) :: _ -> begin
+      match EO.function_slot_offset env function_slot with
+      | Some (Live_function_slot { size; _ }) -> j + size, true
+      | Some Dead_function_slot | None ->
+        (* the function slot was found earlier during the call to
+           order_function_slots *)
         assert false
     end
     | None, (_, Infix_header) :: _ ->
-      (* Cannot happen because a infix header is *always* preceded by a Closure
+      (* Cannot happen because a infix header is *always* preceded by a function
          slot (because the slot list is reversed) *)
       assert false
-    | None, (_, Env_var _) :: _ ->
-      (* Cannot happen because if there is an env var in the acc, then
+    | None, (_, Value_slot _) :: _ ->
+      (* Cannot happen because if there is a value slot in the acc, then
          startenv_opt should be Some _ *)
       assert false
   in
   let slots = List.rev acc_slots in
-  (* The Gc assumes that a set of closures block actually starts with a closure
+  (* The Gc assumes that a Closure_tag block actually starts with a function
      slot at offset 0. Or more precisely, the GC unconditionally reads the
-     second field of a set of closures to find out the start of environment.
-     Thus we add a check here to ensure that the slots start with a closure at
-     offset 0. *)
+     second field of a Closure_tag block to find out the start of environment.
+     Thus we add a check here to ensure that the slots start with a function
+     slot at offset 0. *)
   match slots with
-  | (0, Closure _) :: _ -> { startenv; slots; empty_env }
+  | (0, Function_slot _) :: _ -> { startenv; slots; empty_env }
   | _ ->
     Misc.fatal_error
-      "Sets of closures must start with a closure slot at offset 0"
+      "Sets of function slots must start with a function slot at offset 0"
```

Comment:
In this case, I think it's best to keep "set of closures", or use "block with tag closure_tag"